actionpack/test/dispatch/routing_test.rb

```diff
@@ -4609,6 +4613,17 @@ def show
       assert_response :ok
     end
   end
+
+  test "invalid POST params with a method override cause a bad request" do
+    with_routing do |set|
+      set.draw do
+        patch "/foo/:id", to: "test_invalid_urls/foo#update"
+      end
+
+      post "/foo/1", params: "_method=PATCH&key[]=value&key[0]=value"
```

Comment:
It's a bit unclear from the PR what is meant by invalid params.
I had to read the original issue to find out it is a param that is both an array and a hash.
```suggestion
      invalid_params = "key[]=value&key[0]=value"
      post "/foo/1", params: "_method=PATCH&#{invalid_params}"
```

Reply:
Thanks! Your suggestion it definitely an improvement.

`invalid_params` could basically be anything that `Rack::QueryParser` is unable to parse. It doesn't necessarily have to be an array and a hash. I am going to simplify this to "%", which is a minimal invalid parameter.

Reply:
That's even better 😄